internal/mongo/mongo.go

```diff
@@ -27,12 +22,22 @@ type MongoStorage struct {
 }
 
 func NewMongoStorage(ctx context.Context, opts MongoStorageOptions) (*MongoStorage, error) {
-	client, err := mongo.Connect(ctx, options.Client().ApplyURI(opts.DSN))
 
+	client, err := mongo.NewClient(options.Client().ApplyURI(opts.DSN))
+	if err != nil {
+		return nil, err
+	}
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
```

Comment:
Do you mean the defer or the who context ? 
Connection to database will never timeout if context is removed or used without timeout

Reply:
I mean the defer it should be done one layer higher or not canceled at all because it's just instatiated once it doesn't that much matter if you don't cancel it although the most perfect way is to defer it in the highest level of the code if possible